crates/solidity/testing/sanctuary/src/tests.rs

```diff
@@ -123,3 +138,24 @@ fn extract_compiler_version(compiler: &str) -> Option<Version> {
 
     Some(version)
 }
+
+fn uses_exotic_parser_bug(file: &Path) -> bool {
+    static CONTRACTS_WITH_EXOTIC_PARSER_BUGS: &[&str] = &[
+        // 0.4.24: // Accepts malformed `* /` in multi-line comments:
+        // Fixed in 0.4.25: https://github.com/ethereum/solidity/pull/4937
+        "ethereum/contracts/mainnet/79/79bb6f4492d5cb13fad8ca0ecfbccd9e2c26ac42_Gateway.sol",
+        // 0.4.18: // Accepts unfinished multi-line comments at the end of the file:
+        // Fixed in 0.4.25: https://github.com/ethereum/solidity/pull/4937
+        "ethereum/contracts/mainnet/7d/7d81c361d6ac60634117dd81ab1b01b8dc795a9d_LILITHCOIN.sol",
+        // 0.8.15: Relies on invalidly accepting `indexed indexed` in the event declaration:
+        // Fixed in 0.8.18: https://github.com/ethereum/solidity/pull/13816
+        "ethereum/contracts/mainnet/d4/d4559E5F507eD935F19208A5D50637898c192Ab3_Factory.sol",
+        // 0.4.24: Relies on silently ignoring invalid `\`-escapes in string literals:
```

Comment:
If we can always recover/parse correctly, I wonder if we can do the same here? validating escape sequences and whether they are valid or not can be post-parsing validation. And it doesn’t need to produce an error node, since we correctly tokenized/parsed the string expression.

Reply:
`solc` was silently ignoring invalid escapes; however this means we'd have to modify or escape rules pre 0.4.24 to enable that. If we don't, we incorrectly parse string literals, leading to compilation errors (that we caught during the Sanctuary test suite).

Reply:
Update: We agreed to add a new token definition rule for the single/double quoted string literals that fails to parse with invalid escapes post 0.4.25.

Reply:
Opened https://github.com/NomicFoundation/slang/pull/923